evalml/objectives/utils.py

```diff
@@ -30,6 +30,65 @@ def get_non_core_objectives():
     ]
 
 
+def ranking_only_objectives():
+    """Get ranking-only objective classes.
+
+    Ranking-only objectives are objectives that are useful for evaluating the performance of a model, but should not
+    be used as an optimization objective during AutoMLSearch for various reasons.
+
+    Returns:
+        List of ObjectiveBase classes
+    """
+    return [
```

Comment:
thoughts on adding `ranking_only` as a property instead of maintaining a list? I know this is what we do for `get_non_core_objectives` as well but maybe we could file this as a future enhancement.

Reply:
I'm not quite sure I understand your suggestion - do you mean a property of the objective classes themselves? I do actually like that better, I think.

Reply:
Quick investigation shows that making this switch is unfortunately nontrivial, so I think I'll leave it as is for now. It's a good future enhancement though!